wandb/sdk/internal/handler.py

```diff
@@ -480,28 +479,46 @@ def handle_history(self, record: Record) -> None:
         if updated:
             self._save_summary(self._consolidated_summary)
 
-    def _flush_partial_history(
-        self, partial_history: Optional["PartialHistoryRequest"] = None,
-    ) -> None:
+    def _flush_partial_history(self, step: Optional[int] = None,) -> None:
         if self._partial_history:
             history = HistoryRecord()
             for k, v in self._partial_history.items():
                 item = history.item.add()
                 item.key = k
                 item.value_json = json.dumps(v)
-            if partial_history and partial_history.HasField("step"):
-                history.step.MergeFrom(partial_history.step)
+            if step is not None:
+                history.step.num = step
             self.handle_history(Record(history=history))
             self._partial_history = {}
 
     def handle_request_partial_history(self, record: Record) -> None:
-        history_dict = proto_util.dict_from_proto_list(
-            record.request.partial_history.item
-        )
+        partial_history = record.request.partial_history
+
+        flush = None
+        if partial_history.HasField("action"):
+            flush = partial_history.action.flush
+
+        step = None
+        if partial_history.HasField("step"):
+            step = partial_history.step.num
+
+        history_dict = proto_util.dict_from_proto_list(partial_history.item)
+        if step is not None:
+            if step < self._step:
+                logger.warning(
+                    f"Step {step} < {self._step}. Dropping entry: {history_dict}."
```

Comment:
Just encountered something similar (somewhat) [here](https://github.com/wandb/client/issues/3229#issuecomment-1061498331) (in the `tensorboard` world) - they don't drop it, but record and mark as "outoforder_steps" so that the data are not just dropped. Do you think it's a good idea? Would need additional logic to handle that (as in "ignore"), but at least the data would be preserved.

Also, shouldn't it be `<=`?

Reply:
this logic is just the original logic we had in the user process I just moved it to the internal process.
I guess we could improve the way we deal with steps that are ascending (maybe you can open a ticket for this).

no need for `<=` since if the step matches we just keep updating the current record until we decide to commit it